File: kaleido-gradle-plugin/src/main/java/com/tongsr/kaleido/gradle/AdoptionPlan.java
package com.tongsr.kaleido.gradle;

import java.util.Map;
import java.util.TreeMap;

record AdoptionPlan(Map<String, String> values) {
    AdoptionPlan {
        values = Map.copyOf(new TreeMap<>(values));
    }

    String canonicalText() {
        var sorted = new TreeMap<>(values);
        var text = new StringBuilder();
        sorted.forEach((key, value) -> text.append(key).append('=').append(value).append('\n'));
        return text.toString();
    }
}

File: kaleido-gradle-plugin/src/main/kotlin/com/tongsr/kaleido/gradle/AdoptionPlan.kt
package com.tongsr.kaleido.gradle

@JvmRecord
internal data class AdoptionPlan(val values: Map<String, String>) {
    fun canonicalText(): String = buildString {
        values.toSortedMap().forEach { (key, value) ->
            append(key).append('=').append(value).append('\n')
        }
    }
}
